src/main/java/com/emc/ecs/sync/storage/s3/AwsS3Storage.java

```diff
@@ -83,7 +84,6 @@ public void configure(SyncStorage source, Iterator<SyncFilter> filters, SyncStor
         Assert.hasText(config.getBucketName(), "bucketName is required");
```

Comment:
Lines 81 and 82 above will prevent the other authorization mechanisms added like the default credentials chain or the profile credentials chain because they would require the credentials be passed as part of the payload, even when they are not necessary because they are stored locally on the `ecs-sync` server:

```
Assert.hasText(config.getAccessKey(), "accessKey is required");
Assert.hasText(config.getSecretKey(), "secretKey is required");
```
I would remove those two lines unless the intention is to short circuit those authentication methods in which case a comment would be appropriate.

Reply:
Hi @theronl 
you mean that if we use the STS session way, the access key and secret key will be attached in the `config.getProfile()` ,so `config.getAccessKey()` and `config.getSecretKey()` will be null, right?

Also do you know how to test in the STS session way?

Reply:
The patch includes implementations of session keys as well as the default authentication methods that are part of the AWS SDK.  That way if someone spins up a docker instance with the environment variables set with the AWS credentials or they put in the `~/.sdk/credentials` file the job can be queued simply by setting the following in place of the access key and secret key:
```
<useDefaultCredentialsProvider>true</useDefaultCredentialsProvider>
```
or 
```
<profile>credentials-group</profile>
```
This satisfies the security requirements where the ecs server access is protected but the credentials are not to be passed over the connection

If the assert is kept in place I believe it will prevent these two alternate authentication methods from being used without also including an `accessKey` or `secretKey` in the job request defeating one of the reasons for this patch.

To test in the STS session way you can set up the aws cli with the appropriate credentials.  Here we set up the credentials with the profile group `aws-backup` but if you are using the default credentials you can omit the `--profile <profile-group-name>` part:

```
$ aws --profile aws-backup sts get-session-token
{
    "Credentials": {
        "SecretAccessKey": "XXXXXX",
        "SessionToken": "XXXXXX",
        "Expiration": "2020-04-07T08:50:41Z",
        "AccessKeyId": "XXXXXX"
    }
}
```

The `accessKey` receives the `AccessKeyId` field from the response.
The `secretKey` receives the `SecretAccessKey` field from above.
The `sessionToken` receives the `SessionToken` field from above.

In my example the expiration is set by the IAM.  If you want to override the settings see the [aws cli documentation here[(https://docs.aws.amazon.com/cli/latest/reference/sts/get-session-token.html)